client/src/hooks/reduxHooks.ts

```diff
@@ -0,0 +1,7 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
+import type { RootState, AppDispatch } from '../store';
+
+// Use throughout your app instead of plain `useDispatch` and `useSelector`
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useAppDispatch = () => useDispatch<AppDispatch>();
```

Comment:
This is [the recommended way](https://redux-toolkit.js.org/usage/usage-with-typescript) to make sure `dispatch` and `useSelector` are always properly typed based on our store's known state/actions.